source/solvers/vof_assemblers.cc

```diff
@@ -455,3 +457,162 @@ VOFAssemblerDCDDStabilization<dim>::assemble_rhs(
 
 template class VOFAssemblerDCDDStabilization<2>;
 template class VOFAssemblerDCDDStabilization<3>;
+
+
+template <int dim>
+void
+VOFAssemblerDGCore<dim>::assemble_matrix(
+  const VOFScratchData<dim> &scratch_data,
+  StabilizedMethodsCopyData &copy_data)
+{
+  // Loop and quadrature informations
+  const auto        &JxW_vec    = scratch_data.JxW;
+  const unsigned int n_q_points = scratch_data.n_q_points;
+  const unsigned int n_dofs     = scratch_data.n_dofs;
+
+  // Copy data elements
+  auto &local_matrix = copy_data.local_matrix;
+
+  // assembling local matrix and right hand side
+  for (unsigned int q = 0; q < n_q_points; ++q)
+    {
+      // Gather into local variables the relevant fields
+      const Tensor<1, dim> velocity = scratch_data.velocity_values[q];
+
+      // Store JxW in local variable for faster access;
+      const double JxW = JxW_vec[q];
+
+      for (unsigned int i = 0; i < n_dofs; ++i)
+        {
+          const auto grad_phi_phase_i = scratch_data.grad_phi[q][i];
+
+          for (unsigned int j = 0; j < n_dofs; ++j)
+            {
+              const auto phi_phase_j = scratch_data.phi[q][j];
+
+              // Weak form of : u * grad(phi) =0
+              // Note that the advection term has been weakened for it to appear
+              // explicitly in the weak form as a boundary term.
+              local_matrix(i, j) +=
+                (-grad_phi_phase_i * velocity * phi_phase_j) * JxW;
+            }
+        }
+    } // end loop on quadrature points
+}
+
+
+template <int dim>
+void
+VOFAssemblerDGCore<dim>::assemble_rhs(const VOFScratchData<dim> &scratch_data,
+                                      StabilizedMethodsCopyData &copy_data)
+{
+  // Loop and quadrature informations
+  const auto        &JxW_vec    = scratch_data.JxW;
+  const unsigned int n_q_points = scratch_data.n_q_points;
+  const unsigned int n_dofs     = scratch_data.n_dofs;
+
+  // Copy data elements
+  auto &local_rhs = copy_data.local_rhs;
+
+  // Assembling local right hand side
+  for (unsigned int q = 0; q < n_q_points; ++q)
+    {
+      // Gather into local variables the relevant fields
+      const double         phase_value = scratch_data.present_phase_values[q];
+      const Tensor<1, dim> velocity    = scratch_data.velocity_values[q];
+
+      // Store JxW in local variable for faster access;
+      const double JxW = JxW_vec[q];
+
+      for (unsigned int i = 0; i < n_dofs; ++i)
+        {
+          const auto grad_phi_phase_i = scratch_data.grad_phi[q][i];
+
+          // Weak form of : u * grad(phi) =0
```

Comment:
Do you still assemble to use the non-linear solver? If so, I would write (also at line 493):

```suggestion
          // Linearized weak form of the strong problem u * grad(phi) =0
```